metagraph/src/cli/stats.cpp

```diff
@@ -181,6 +181,40 @@ void print_brwt_stats(const annot::binmat::BRWT& brwt) {
 }
 
 void print_annotation_stats(const std::string &fname, const Config &config) {
+    if (config.print_column_names) {
+        annot::LabelEncoder<std::string> label_encoder;
+
+        logger->info("Scanning annotation '{}'", fname);
+
+        try {
+            std::ifstream instream(fname, std::ios::binary);
+
+            // TODO: make this more reliable
+            if (parse_annotation_type(fname) == Config::ColumnCompressed) {
+                // Column compressed dumps the number of rows first
+                // skipping it...
+                load_number(instream);
+            }
+
+            if (!label_encoder.load(instream))
+                throw std::ios_base::failure("");
+
+        } catch (...) {
+            logger->error("Cannot read label encoder from file '{}'", fname);
+            exit(1);
+        }
+
+        std::cout << "Number of columns: " << label_encoder.size() << '\n';
+        for (size_t c = 0; c < label_encoder.size(); ++c) {
+            std::cout << label_encoder.decode(c) << '\n';
+        }
+
+        std::cout << std::flush;
```

Comment:
ColumnCompressed additionally stores the number of rows, so it must be handled separately in any case

Reply:
My point is that if you force the annotation to be loaded using mmap (regardless of user set flags), then all you would have to do here is simply load the annotation, then use API calls to output the metadata you need.

Reply:
This code relies too much on the implementation details of the annotations, so it would be nice if we could avoid that

Reply:
Do you mean always load annotations with `mmap` in `stats`? It's would be slower because loading thousands of columns with mmap has a noticeable overhead, so I'd rather avoid that.

Reply:
ah, ok then. Thanks for explaining